experanto/datasets.py

```diff
@@ -0,0 +1,448 @@
+                # for the screen, we'll utilize torchvision.transforms
+                add_channel = Lambda(lambda x: torch.from_numpy(x[:, None, ...]) if len(x.shape) == 3 else torch.from_numpy(x))
+                transform_list = [v for v in self.modality_config.screen.transforms.values()]
+                transform_list.insert(0, add_channel)
+                transforms[device_name] = Compose(transform_list)
+            else:
+                if self.modality_config[device_name].transforms:
+                    transforms[device_name] = Compose([
+                        ToTensor(),
+                        torchvision.transforms.Normalize(self._statistics[device_name]["mean"], self._statistics[device_name]["std"])])
+                else:
+                    transforms[device_name] = Compose([
+                        ToTensor()])
+        return transforms
+
+    def get_sample_in_meta_condition(self) -> dict:
+        """
+        iterates through all
+           for example:
+              if meta_conditions = {"tier": [train,train, ...], "stim_type": [type1, type2, ...]}
+              and valid_condition = {"tier": train, "stim_type": type2}
+           then the output would be {"tier": [True, True, ...], "stim_type": [False, True, ...]}
+        """
+        sample_in_meta_condition = {}
+        for k, v in self.modality_config["screen"]["valid_condition"].items():
+            sample_in_meta = []
+            for i, (condition, start, end) in enumerate(zip(self.meta_conditions[k], self._start_times, self._end_times)):
+                if condition == v or (condition == "blank" and self.modality_config["screen"]["include_blanks"]):
+                    sample_in_meta.append(
+                        (self._screen_sample_times >= start) & (self._screen_sample_times < end)
+                    )
+            sample_in_meta_condition[k] = np.stack(sample_in_meta).sum(0).astype(bool)
+        return sample_in_meta_condition
+
+    def get_full_valid_sample_times(self, ) -> Iterable:
+        """
+        iterates through all sample times and checks if they are in the meta condition of interest
+        :return:
+        """
+        valid_times = []
+        for i, _ in enumerate(self._screen_sample_times[:-self.chunk_sizes["screen"]]):
+            maybe_all_true = []
+            correct_duration = self._screen_sample_times[i + self.chunk_sizes["screen"]] < self.end_time
+            maybe_all_true.append(correct_duration)
+            for k, v in self._sample_in_meta_condition.items():
+                maybe_all_true.append(np.all(v[i: i + self.chunk_sizes["screen"]]))
+            if np.all(maybe_all_true):
+                valid_times.append(self._screen_sample_times[i])
+        return np.stack(valid_times)
+
+    def shuffle_valid_screen_times(self) -> None:
+        """
+        convenience function to randomly select new starting points for each chunk. Use this in training after each epoch.
```

Comment:
maybe its a nice idea to add more comments (from slack with Mara) on why this is here

Reply:
Yes agreed. I thought that this is a good idea to add this function, but for all practical purposes, the training should be `stride=1` anyhow got get as much data as possible. so now I'm thinking that this function isn't super useful, but might have its use cases